emmet/borg/icsd_to_mongo.py

```diff
@@ -0,0 +1,255 @@
+    	bibtex_str += "}"
+    	#bibtex_str = unicodedata.normalize('NFKD', bibtex_str).encode('ascii','ignore')
+
+    	#print(bibtex_str)
+
+    	return bibtex_str
+
+
+    	return cif_string
+
+class icsdQueen(BorgQueen):
+
+    def __init__(self, drone, rootpath=None, number_of_drones=1):
+        self._drone = drone
+        self._num_drones = number_of_drones
+        self._data = []
+
+        if rootpath:
+            if number_of_drones > 1:
+                self.parallel_assimilate(rootpath)
+            else:
+                self.serial_assimilate(rootpath)
+
+    def parallel_assimilate(self, rootpath):
+        """
+        Assimilate the entire subdirectory structure in rootpath.
+        """
+        logger.info('Scanning for valid paths...')
+        valid_paths = []
+        for (parent, subdirs, files) in os.walk(rootpath):
+            valid_paths.extend(self._drone.get_valid_paths(self._drone, (parent, subdirs,
+                                                            files)))
+        manager = Manager()
+        data = manager.list()
+        status = manager.dict()
+        status['count'] = 0
+        status['total'] = len(valid_paths)
+        logger.info('{} valid paths found.'.format(len(valid_paths)))
+        p = Pool(self._num_drones)
+        p.map(self.order_assimilation, ((path, self._drone, data, status)
+                                   for path in valid_paths))
+        for d in data:
+            self._data.append(json.loads(d, cls=MontyDecoder))
+
+    def serial_assimilate(self, rootpath):
+        """
+        Assimilate the entire subdirectory structure in rootpath serially.
+        """
+        valid_paths = []
+        for (parent, subdirs, files) in os.walk(rootpath):
+            valid_paths.extend(self._drone.get_valid_paths(self._drone, (parent, subdirs,
+                                                            files)))
+        data = []
+        count = 0
+        total = len(valid_paths)
+        for path in valid_paths:
+            newdata = self._drone.assimilate(self._drone, path)
+            self._data.append(newdata)
+            count += 1
+            logger.info('{}/{} ({:.2f}%) done'.format(count, total,
+                                                      count / total * 100))
+        for d in data:
+            self._data.append(json.loads(d, cls=MontyDecoder))
+
+    def order_assimilation(self, args):
+        """
+        Internal helper method for BorgQueen to process assimilation
+        """
+        (path, drone, data, status) = args
+        newdata = drone.assimilate(drone, path)
+        if newdata:
+            data.append(json.dumps(newdata, cls=MontyEncoder))
+        status['count'] += 1
+        count = status['count']
+        total = status['total']
+        logger.info('{}/{} ({:.2f}%) done'.format(count, total,
+                                                  count / total * 100))
+
+rootpath = '/home/jpalaks/Documents/MP_Work/Scraped Files'
```

Comment:
You can default `rootpath` to the current working directory or something. Also, wrap it and the invocation below in a

    if __name__ == '__main__':

block so it doesn't get invoked when just importing this module.